esmvalcore/cmor/check.py

```diff
@@ -64,7 +64,7 @@ def __init__(self,
         self._errors = list()
         self._warnings = list()
         self._cmor_var = var_info
-        if frequency is None:
+        if not frequency:
```

Comment:
To include also frequency being an empty string

Reply:
I suggest the more explicit `frequency in [None, '']` then. This is less prone to future breakage and clearer to read.

Perhaps an even better way to address it would be to standardize on *one* singleton, either `None` or `''`, to mark absence consistently, but I am not sure how far reaching and feasible that would be.

Reply:
From PEP8:

> For sequences, (strings, lists, tuples), use the fact that empty sequences are false.
Yes: if not seq:
     if seq:
No:  if len(seq):
     if not len(seq):

We are doing this several times across the whole tool

Reply:
The goal here is not to check the length of a sequence, but to compare with exactly two singletons. The fact that one of them is a sequence is purely coincidental, so imho the section you quote is not applicable here.

The risk I see for the future is twofold: At some point someone might see this code together with the `None` default argument a few lines above and decide to refactor it into `if frequency is None:`. The other thing that is a possible different value that evaluates to false, perhaps an empty list or some form of numerical zero in a new table?

Having said all that, I did approve this PR, so take it as a recommendation and feel free to ignore it if you really come down on the other side of the fence.

Reply:
@jvegasbsc guess you want to keep it that way. Resolving as a matter of procedure.